nixos/modules/services/security/oauth2_proxy.nix

```diff
@@ -579,7 +579,7 @@ in
       description = "OAuth2 Proxy";
       path = [ cfg.package ];
       wantedBy = [ "multi-user.target" ];
-      after = [ "network.target" ];
+      after = [ "network-online.target" ];
```

Comment:
This also needs a `wants  = [ "network-online.target" ]`  as per https://systemd.io/NETWORK_ONLINE/

If `oauth2_proxy` is the only service that is ordered after `network-online.target` nothing will happen as `network-online.target` is a _passive_ unit.

Reply:
Ah this will be fixed by https://github.com/NixOS/nixpkgs/pull/258680